src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileVersionProvider.cs

```diff
@@ -76,35 +77,39 @@ public string AddFileVersionToPath(string path)
                 return path;
             }
 
-            var fileInfo = _fileProvider.GetFileInfo(resolvedPath);
-            if (!fileInfo.Exists)
+            string value;
+            if (!_cache.TryGetValue(path, out value))
             {
-                if (_requestPathBase.HasValue &&
+                var cacheEntryOptions = new MemoryCacheEntryOptions();
+                var fileInfo = _fileProvider.GetFileInfo(resolvedPath);
+                cacheEntryOptions.AddExpirationToken(_fileProvider.Watch(resolvedPath));
+
+                if (!fileInfo.Exists &&
+                    _requestPathBase.HasValue &&
                     resolvedPath.StartsWith(_requestPathBase.Value, StringComparison.OrdinalIgnoreCase))
                 {
                     resolvedPath = resolvedPath.Substring(_requestPathBase.Value.Length);
                     fileInfo = _fileProvider.GetFileInfo(resolvedPath);
+                    cacheEntryOptions.AddExpirationToken(_fileProvider.Watch(resolvedPath));
```

Comment:
Up one, confused me at firsti because I thought you were doing the same thing twice. WHen in reality you're recaclculating the resolvedPath.